Mathlib/FieldTheory/SeparableDegree.lean

```diff
@@ -18,78 +18,112 @@ This file contains basics about the separable degree of a field extension.
 
 ## Main definitions
 
-- `Field.Emb F E`: the type of `F`-algebra homomorphisms from `E` to the algebraic closure of `E`.
+- `Field.Emb F E`: the type of `F`-algebra homomorphisms from `E` to the algebraic closure of `E`
+  (mathematically, it should be the algebraic closure of `F`, but in order to make its cardinality
+  has the same type with `Module.rank F E`, we use the algebraic closure of `E`).
   Usually denoted by $\operatorname{Emb}_F(E)$ in textbooks.
 
-- `Field.sepDegree F E`: the separable degree $[E:F]_s$ of an algebraic extension `E / F` of fields,
-  defined to be the cardinal of `F`-algebra homomorphisms from `E` to the algebraic closure of `E`.
-  (Mathematically, it should be the algebraic closure of `F`, but in order to make the type
-  compatible with `Module.rank F E`, we use the algebraic closure of `E`.)
+  **Remark:** if `E / F` is not algebraic, then this definition makes no mathematical sense,
+  and if it is infinite, then it behaves badly. For example,
+  $\operatorname{Hom}_{\mathbb{Q}\text{-alg}}(\mathbb{Q}(\mu_{p^\infty}),
+  \overline{\mathbb{Q}})\cong\mathbb{Z}_p^\times$ which is uncontable, while
+  $[\mathbb{Q}(\mu_{p^\infty}):\mathbb{Q}]$ is countable.
```

Comment:
When you say "behaves badly" I think you originally mean that the cardinality #(Emb F E) doesn't behave as expected when E/F is infinite, not `Emb F E` itself. But even for the cardinality, I wouldn't say it behaves badly, because it still behaves quite regularly (equal to 2^deg (sep closure), I believe); it's just the typical behavior of automorphism groups of infinite objects.

Reply:
You're right, I rephrased it a little bit. But I didn't mention 2^deg (sep closure) thing.